src/backend/tcop/postgres.c

```diff
@@ -5314,6 +5336,24 @@ PostgresMain(int argc, char *argv[], const char *username)
 	return 1;					/* keep compiler quiet */
 }
 
+static void
+curl_finalize(int code, Datum arg __attribute__((unused)))
+{
+	if (curl_context.hasInited)
+	{
+		if (curl_context.response.buffer != NULL) {
+			pfree(curl_context.response.buffer);
+		}
+		/* cleanup curl stuff */
+		if (curl_context.curl_handle) {
+			curl_easy_cleanup(curl_context.curl_handle);
+		}
+		/* we're done with libcurl, so clean it up */
+		curl_global_cleanup();
+		curl_context.hasInited = false;
+		elog(LOG, "finalize the global struct for curl handle context.");
```

Comment:
Agreed with paul, maybe `DEBUG5` is ok.
But how to adjust log level of hawq? @paul-guo-  And `DEBUG5` log is not writen in log file now.